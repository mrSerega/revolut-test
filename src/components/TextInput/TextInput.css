.text-input {
    position: relative;
    display: flex;
    justify-content: flex-end;
}

.text-input__view,
.text-input__input {
    background-color: transparent;
    outline: none;
    border: none;
    color: transparent;
    font-size: 32px;
    width: 100%;
    text-align: right;
}

.text-input__input {
    color: white;
    z-index: 1;
    position: relative;
}

.text-input__view {
    position: absolute;
    color: transparent;
    z-index: 0;
}

.text-input__view:not(.text-input__view_empty)::before {
    content: "-";
    color: #ffffff;
}

.text-input__placeholder {
    z-index: 0;
    position: absolute;
    height: 100%;
    width: 100%;
    top: 0;
    left: 0;
    text-align: right;
    font-size: 32px;
    color: rgba(255,255,255,0.25)
}