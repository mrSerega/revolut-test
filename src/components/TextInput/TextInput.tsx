import React, { ChangeEvent } from 'react'

import './TextInput.css'

export interface TextInputProps {
    value: string
    onChange: (value: string) => void;
    tabIndex?: number
    autoFocus?: boolean;
    placeholder?: string
    isFrom?: boolean
}

const floatRegExp = /^(\d*)\.?(\d{1,2})?$/

export class TextInput extends React.Component<TextInputProps> {
    readonly blockName = 'text-input'

    render() {

        const {
            value,
            tabIndex,
            autoFocus = false,
            placeholder,
            isFrom
        } = this.props

        return <div className={this.blockName}>
            {isFrom && <div className={
                this.blockName + '__view' + ' ' +
                this.blockName + '__view' + (!value ? '_empty' : '')
            }>
                {value}
            </div>}
            <input
                autoFocus={autoFocus}
                tabIndex={tabIndex}
                type="text"
                className={this.blockName + '__input'}
                value={value || ''}
                onChange={this.handleChange}
            />
            <div className={this.blockName + '__placeholder'}>
                {!value ? placeholder : ""}
            </div>
        </div>
    }

    private handleChange = (evt: ChangeEvent<HTMLInputElement>) => {
        const {
            onChange
        } = this.props

        const val = evt.target.value

        if (floatRegExp.test(val)) {
            onChange(val)
        }
    }
}