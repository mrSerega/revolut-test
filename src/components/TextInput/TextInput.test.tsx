import React from 'react';
import renderer from 'react-test-renderer';
import { TextInput } from './TextInput';

describe ('TextInput', () => {
    test('With empty value and placeholder', () => {
        const cmp = renderer.create(
            <TextInput
                value=""
                onChange={() => {}}
                placeholder="placeholder"
            />
        )
        const tree = cmp.toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('With not empty value and placeholder', () => {
        const cmp = renderer.create(
            <TextInput
                value="100"
                onChange={() => {}}
                placeholder="placeholder"
            />
        )
        const tree = cmp.toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('With not empty value and without placeholder', () => {
        const cmp = renderer.create(
            <TextInput
                value="100"
                onChange={() => {}}
            />
        )
        const tree = cmp.toJSON()
        expect(tree).toMatchSnapshot()
    })
    test('With isFrom', () => {
        const cmp = renderer.create(
            <TextInput
                isFrom
                value="100"
                placeholder="placeholder"
                onChange={() => {}}
            />
        )
        const tree = cmp.toJSON()
        expect(tree).toMatchSnapshot()
    })
})
