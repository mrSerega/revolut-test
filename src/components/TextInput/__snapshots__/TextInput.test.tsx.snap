// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput With empty value and placeholder 1`] = `
<div
  className="text-input"
>
  <input
    autoFocus={false}
    className="text-input__input"
    onChange={[Function]}
    type="text"
    value=""
  />
  <div
    className="text-input__placeholder"
  >
    placeholder
  </div>
</div>
`;

exports[`TextInput With isFrom 1`] = `
<div
  className="text-input"
>
  <div
    className="text-input__view text-input__view"
  >
    100
  </div>
  <input
    autoFocus={false}
    className="text-input__input"
    onChange={[Function]}
    type="text"
    value="100"
  />
  <div
    className="text-input__placeholder"
  >
    
  </div>
</div>
`;

exports[`TextInput With not empty value and placeholder 1`] = `
<div
  className="text-input"
>
  <input
    autoFocus={false}
    className="text-input__input"
    onChange={[Function]}
    type="text"
    value="100"
  />
  <div
    className="text-input__placeholder"
  >
    
  </div>
</div>
`;

exports[`TextInput With not empty value and without placeholder 1`] = `
<div
  className="text-input"
>
  <input
    autoFocus={false}
    className="text-input__input"
    onChange={[Function]}
    type="text"
    value="100"
  />
  <div
    className="text-input__placeholder"
  >
    
  </div>
</div>
`;
